core/src/main/java/org/mule/runtime/core/processor/chain/ModuleOperationMessageProcessorChainBuilder.java

```diff
@@ -0,0 +1,134 @@
+ *    <module-operation-properties/>
+ *    <module-operation-parameters>
+ *      <module-operation-parameter-entry value="howdy" key="value1"/>
+ *      <module-operation-parameter-entry value="#[flowVars.person]" key="value2"/>
+ *    </module-operation-parameters>
+ *    <set-payload value="#[param.value1 + param.value2]"/>
+ * </module-operation-chain>
+ * </pre>
+ */
+public class ModuleOperationMessageProcessorChainBuilder extends ExplicitMessageProcessorChainBuilder {
+
+  private Map<String, String> properties;
+  private Map<String, String> parameters;
+  private boolean returnsVoid;
+  private ExpressionLanguage expressionLanguage;
+
+  public ModuleOperationMessageProcessorChainBuilder(Map<String, String> properties,
+                                                     Map<String, String> parameters, boolean returnsVoid,
+                                                     ExpressionLanguage expressionLanguage) {
+    this.properties = properties;
+    this.parameters = parameters;
+    this.returnsVoid = returnsVoid;
+    this.expressionLanguage = expressionLanguage;
+  }
+
+  @Override
+  protected MessageProcessorChain createInterceptingChain(Processor head, List<Processor> processors,
+                                                          List<Processor> processorForLifecycle) {
+    return new ModuleOperationProcessorChain("wrapping-operation-module-chain", head, processors, processorForLifecycle,
+                                             properties, parameters, returnsVoid,
+                                             expressionLanguage);
+  }
+
+  /**
+   * Generates message processor for a specific set of parameters & properties to be added in a new event.
+   */
+  static class ModuleOperationProcessorChain extends ExplicitMessageProcessorChain
+      implements Processor, MessageProcessorContainer {
+
+    private Map<String, String> properties;
+    private Map<String, String> parameters;
+    private boolean returnsVoid;
+    private ExpressionLanguage expressionLanguage;
+
+    ModuleOperationProcessorChain(String name, Processor head, List<Processor> processors,
+                                  List<Processor> processorsForLifecycle,
+                                  Map<String, String> properties, Map<String, String> parameters,
+                                  boolean returnsVoid,
+                                  ExpressionLanguage expressionLanguage) {
+      super(name, head, processors, processorsForLifecycle);
+      this.properties = properties;
+      this.parameters = parameters;
+      this.returnsVoid = returnsVoid;
+      this.expressionLanguage = expressionLanguage;
+    }
+
+    @Override
+    public void addMessageProcessorPathElements(MessageProcessorPathElement pathElement) {
+      MessageProcessorPathElement subprocessors = pathElement.addChild(name).addChild("subprocessorsModuleOperations");
+      NotificationUtils.addMessageProcessorPathElements(processors, subprocessors);
```

Comment:
can't, the static method has the same signature as the current function, hence cannot redefine it by static import